packages/components/src/components/block/block.tsx

```diff
@@ -586,9 +609,16 @@ export class Block extends LitElement implements InteractiveComponent {
         ) : (
           headerContent
         )}
-        {iconEnd && !collapsible ? (
-          <div class={CSS.iconEndContainer}>{this.renderIcon("end")}</div>
-        ) : null}
+        {(() => {
```

Comment:
Can you remove the iffe? This shouldn't be necessary and ends up creating a new anonymous function on each render:

```tsx
{
  iconEnd && !collapsible ? (
    <div class={CSS.iconEndContainer}>
      {this.renderContentEnd()}
      {this.renderIcon("end")}
    </div>
  ) : !iconEnd && !collapsible ? (
    this.renderContentEnd()
  ) : null
}
```